templates/common/openstack/files/usr-local-bin-openstack-kubelet-nodename.yaml

```
@@ -31,16 +31,33 @@ contents:
     # Set node name to be instance name instead of the default FQDN hostname.
     # Try to get the hostname from config drive first if it's available
     # or fallback to the metadata service.
-    if [ -f "/host/var/config/openstack/2012-08-10/meta_data.json" ]; then
+    if [ ! -z "$(blkid -t LABEL="config-2" -odevice)" ]; then
```

nit:

```suggestion
if [ -n "$(blkid -t LABEL="config-2" -odevice)" ]; then
```

Same thing, no?